### PR TITLE
Add ore bag area pickups

### PR DESCRIPTION
**Pull request by metalgearsloth**
Apparently it does it in 13.

Update: Needs https://github.com/space-wizards/space-station-14/pull/19682

https://github.com/space-wizards/space-station-14/assets/31366439/aedc4ed3-4cb0-4a18-a112-63bd5309befa

:cl:
- add: Added automatic area pickups to ore bags.
- fix: Fix stackable items not playing the pickup animation on pickup.

### CODE REVIEW

**Comment by deltanedas** [comment]
fixes #17949 :trollface:

**Comment by metalgearsloth** [comment]
> fixes #17949 :trollface:

noob

**Comment by deltanedas** [comment]
no, pro

File: Content.Server/Storage/EntitySystems/MagnetPickupSystem.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+/// <see cref="MagnetPickupComponent"/>
+/// </summary>
+public sealed class MagnetPickupSystem : EntitySystem
+{
+    [Dependency] private readonly IGameTiming _timing = default!;
+    [Dependency] private readonly EntityLookupSystem _lookup = default!;
+    [Dependency] private readonly InventorySystem _inventory = default!;
+    [Dependency] private readonly StorageSystem _storage = default!;
+
+    private static readonly TimeSpan ScanDelay = TimeSpan.FromSeconds(1);
```
Comment (mirrorcult) [inline Content.Server/Storage/EntitySystems/MagnetPickupSystem.cs:18]: datafield?

**Review by mirrorcult — DISMISSED** [review]
doesnt do the pickup animation? i think it might not be obvious enough whats happening. should probably have a description change as well + maybe guidebook

**Comment by metalgearsloth** [comment]
> doesnt do the pickup animation? i think it might not be obvious enough whats happening. should probably have a description change as well + maybe guidebook

Do you have any suggestions as item pickup animations were not built to handle entities being deleted the same tick.

File: Content.Server/Storage/EntitySystems/MagnetPickupSystem.cs (hunk 1 above)
Comment (metalgearsloth) [reply Content.Server/Storage/EntitySystems/MagnetPickupSystem.cs:18]: sur

**Comment by deltanedas** [comment]
could that not spawn a client copy and then run animation on that, deleting it when its over

**Comment by metalgearsloth** [comment]
> could that not spawn a client copy and then run animation on that, deleting it when its over

That is how pickup animations work but the code is scrungly bungly and I am like 60% sure will just die because the entities are immediately deleted (i.e. where stack stuff is picked up).

**Comment by metalgearsloth** [comment]
Gonna need to refactor storagesystem entirely for this to be predicted so will come back to itTM.

**Comment by metalgearsloth** [comment]
> doesnt do the pickup animation? i think it might not be obvious enough whats happening. should probably have a description change as well + maybe guidebook

I did all that except guidebook coz I've never done it and uhhhh not sure where it'd go.

**Pull request merged**